types/unist/index.d.ts

```diff
@@ -79,19 +80,22 @@ export interface Point {
     offset?: number | undefined;
 }
 
+export type NodeData<TNode extends Node<object>> = TNode extends Node<infer NData> ? NData : never;
+
 /**
  * Nodes containing other nodes.
  */
-export interface Parent extends Node {
+export interface Parent<ChildNode extends Node<object> = Node, NData extends object = NodeData<ChildNode>>
```

Comment:
Not sure if it will occur here, but TypeScript has depth limitations when checking `extends`, so it's _possible_ (correct me if I'm wrong) that `NodeData` will actually fail in some cases (as `NodeData` uses `extends`). I just tried it in playground with quite a few occurrences of `Parent<Parent<Parent<...<Node>...>` and it seems to be working okay, though.

Reply:
The max depth limit should be 50 as I remember, and I don’t think it will occur in practise. Who will write `Parent<Parent<Parent<...<Node>...>` with 50 depth?

Reply:
Typings can get complex pretty quickly, and TS borks pretty easily too. That said, because the default for this shouldn't hit this, there should be no breakage as a result.

Reply:
Yep, I can't see any possible it will happen in practice.